packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -1060,3 +1125,143 @@ contract OptimismPortalResourceFuzz_Test is CommonTest {
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(1), 0, 0, gasLimit, true, "");
+    }
+
+    /// @dev Tests that `depositERC20Transaction` reverts when the gas limit is too low.
+    function test_depositERC20Transaction_gasLimitTooLow_reverts() external {
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        vm.expectRevert("OptimismPortal: gas limit too small");
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), 0, 0, 0, false, "");
+    }
+
+    /// @dev Tests that `depositERC20Transaction` reverts when the data is too large.
+    function test_depositERC20Transaction_dataTooLarge_reverts() external {
+        bytes memory data = new bytes(120_001);
+        data[120_000] = 0x01;
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        uint64 gasLimit = optimismPortal.minimumGasLimit(120_001);
+
+        vm.expectRevert("OptimismPortal: data too large");
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), 0, 0, gasLimit, false, data);
+    }
+
+    /// @dev Tests that `depositERC20Transaction` reverts when not enough of the token is approved.
+    function test_depositERC20Transaction_notEnoughAmount_reverts() external {
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+        vm.expectRevert();
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), 1, 0, 0, false, "");
+    }
+
+    /// @dev Tests that `balance()` returns the correct balance when the gas paying token is not ether.
+    function testFuzz_balance_nonEther_succeeds(uint256 _amount) external {
+        // Mint the token to the contract and approve the token for the portal
+        token.mint(address(this), _amount);
+        token.approve(address(optimismPortal), _amount);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), _amount, 0, optimismPortal.minimumGasLimit(0), false, "");
+
+        // Check that the balance has been correctly updated
+        assertEq(optimismPortal.balance(), _amount);
+    }
+
+    /// @dev Tests that `finalizeWithdrawalTransaction` succeeds.
+    function test_finalizeWithdrawalTransaction_provenWithdrawalHash_nonEther_succeeds(bytes memory _data) external {
+        _defaultTx.data = _data;
```

Comment:
i am confused how you can malleate the `_defaultTx` and still have this test pass, since it will invalidate its inclusion in the output root and it should be an invalid withdrawal proof

Reply:
you're totally right. was just seeing this issue upon running tests locally